test/NuGet.Core.Tests/NuGet.Common.Test/RuntimeEnvironmentHelperTests.cs

```diff
@@ -13,10 +13,11 @@ namespace NuGet.Common.Test
     public class RuntimeEnvironmentHelperTests
     {
         [Fact]
-        public void RuntimeEnvironmentHelper_MutuallyExclusiveRuntimes()
+        public void RuntimeEnvironmentHelper_OperatingSystemChecks_OnlyOneReturnsTrue()
```

Comment:
A simpler fix for release-5.0.x would be to undo changes to this file.  Then open a new issue tracking the `IsMacOSX`/`IsLinux` bug.  Then open a new PR (targeting only the `dev` branch) that adds test disabled with a link to the bug you just opened.